### PR TITLE
Add a dummy `beginMarkedContentProps` operator when optional content parsing fails (issue 17679)

### LINKED ISSUE #17679: Corrupt layer definition make pdf.js fail to render a pdf
Attach (recommended) or Link to PDF file here:
[layersMissingDefs.pdf](https://github.com/mozilla/pdf.js/files/14310187/layersMissingDefs.pdf)

Note: this pdf is not the original PDF, but boiled down to the offending issue. The original document was produced by merging two pdfs, where one contained layers. Each pdf was placed into a new layer, but the creator application failed to include the original layer definitions. The resulting pdf is arguably broken. Acrobat can handle it nevertheless. Also, I believe the current approach to handle malformed marked content is not ideal and deserves revisiting independent of this concrete issue (see technical assessment).

Configuration:

- Web browser and its version: Firefox 122.0.1
- Operating system and its version: Windows 22H2 (19045.3930)
- PDF.js version: PDF.js: 4.0.348 [cb009d15a]
- Is a browser extension: built-in in Firefox

Steps to reproduce the problem:
1. open the attached pdf in Firefox
2. open the layer panel
3. toggle visibility of layer "green" - works fine
4. toggle visibility of layer "red" - rendering is not updated, but spinner shows

What is the expected behavior? (add screenshot)
image

What went wrong? (add screenshot)
image

Firefox console output:
PDF 4a98efbce201625d6366fd6b63e1ee86 [1.4 Notepad / Notepad] (PDF.js: 4.0.348 [cb009d15a]) [viewer.mjs:10603:13](resource://pdf.js/web/viewer.mjs)
Warning: getOperatorList - ignoring beginMarkedContentProps: "TypeError: properties is undefined". 2 [pdf.worker.mjs:342:13](resource://pdf.js/build/pdf.worker.mjs)
Uncaught (in promise) TypeError: this.ctx is undefined
    restore resource://pdf.js/build/pdf.mjs:6089
    #restoreInitialState resource://pdf.js/build/pdf.mjs:5798
    endDrawing resource://pdf.js/build/pdf.mjs:5811
    cancel resource://pdf.js/build/pdf.mjs:10195
    promise callback*_scheduleNext/< resource://pdf.js/build/pdf.mjs:10224
    _scheduleNext resource://pdf.js/build/pdf.mjs:10223
    renderContinueCallback resource://pdf.js/web/viewer.mjs:7421
    _continue resource://pdf.js/build/pdf.mjs:10216
    operatorListChanged resource://pdf.js/build/pdf.mjs:10208
    render resource://pdf.js/build/pdf.mjs:9056
    promise callback*render resource://pdf.js/build/pdf.mjs:9046
    draw resource://pdf.js/web/viewer.mjs:7474
    renderView resource://pdf.js/web/viewer.mjs:4697
    forceRendering resource://pdf.js/web/viewer.mjs:8644
    promise callback*forceRendering resource://pdf.js/web/viewer.mjs:8643
    renderHighestPriority resource://pdf.js/web/viewer.mjs:4627
    update resource://pdf.js/web/viewer.mjs:8540
    refresh resource://pdf.js/web/viewer.mjs:9043
    set optionalContentConfigPromise resource://pdf.js/web/viewer.mjs:8704
    webViewerOptionalContentConfig resource://pdf.js/web/viewer.mjs:11216
    dispatch resource://pdf.js/web/viewer.mjs:916
    setVisibility resource://pdf.js/web/viewer.mjs:3915
    onclick resource://pdf.js/web/viewer.mjs:3922
    _bindLink resource://pdf.js/web/viewer.mjs:3920
    render resource://pdf.js/web/viewer.mjs:3996
    load resource://pdf.js/web/viewer.mjs:10520
    promise callback*load/< resource://pdf.js/web/viewer.mjs:10516
    promise callback*load resource://pdf.js/web/viewer.mjs:10489
    open resource://pdf.js/web/viewer.mjs:10250
    promise callback*open resource://pdf.js/web/viewer.mjs:10249
    onOpenWithData resource://pdf.js/web/viewer.mjs:10120
    windowMessage resource://pdf.js/web/viewer.mjs:12160
    initPassiveLoading resource://pdf.js/web/viewer.mjs:12121
    initPassiveLoading resource://pdf.js/web/viewer.mjs:10110
    run resource://pdf.js/web/viewer.mjs:10042
    webViewerLoad resource://pdf.js/web/viewer.mjs:12482
    <anonymous> resource://pdf.js/web/viewer.mjs:12486
[pdf.mjs:6089:9](resource://pdf.js/build/pdf.mjs)

​Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

Technical assessment: The issue is that pdf.js skips `OPS.beginMarkedContentProps` that somehow are malformed when parsing a pdf (https://github.com/mozilla/pdf.js/blob/master/src/core/evaluator.js#L2184) but corresponding `OPS.endMarkedContent` are accepted. This makes marked content begin/ends unbalanced with unknown consequences. It would be better to either not skip the begin or also skip the end to keep the balance.

### PR DESCRIPTION
**Pull request by Snuffleupagus**

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -2204,6 +2204,10 @@ class PartialEvaluator {
                       warn(
                         `getOperatorList - ignoring beginMarkedContentProps: "${reason}".`
                       );
+                      operatorList.addOp(OPS.beginMarkedContentProps, [
+                        "OC",
+                        null,
+                      ]);
```
Comment (haikl) [inline src/core/evaluator.js:2210]: Thanks for the quick reaction! Maybe add a dummy op after line 2186 as well.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**